src/Proto3/Suite/Form/Encode/Core.hs

```diff
@@ -354,13 +355,13 @@ class Field name a message
     field :: forall names . a -> Prefix message names (Occupy message name names)
 
 instance forall (name :: Symbol)
-#if MIN_VERSION_ghc(9,4,0)
-                r (a :: TYPE r)
-#else
+#if defined(__GLASGOW_HASKELL__) && __GLASGOW_HASKELL__ < 904
```

Comment:
Not sure if it matters in practice, but I think if the macro is not defined then we are unlikely to have support for the necessary runtime polymorphism, presumably because the compiler is not GHC.

Reply:
I did this in case you were thinking of compilers other than GHC, but you're right, you're right.